llama-index-integrations/embeddings/llama-index-embeddings-deepinfra/llama_index/embeddings/deepinfra/base.py

```diff
@@ -0,0 +1,200 @@
+        """
+        Sends a POST request to the DeepInfra Inference API with the given data and returns the API response.
+        Input data is chunked into batches to avoid exceeding the maximum batch size (1024).
+
+        Args:
+            data (List[str]): A list of strings to be embedded.
+        Output:
+            List[float]: A list of embeddings from the API.
+
+        """
+        url = self.get_url()
+        chunked_data = _chunk(data)
+        embeddings = []
+        for chunk in chunked_data:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(
+                    url,
+                    json={
+                        "inputs": chunk,
+                    },
+                    headers={
+                        "Authorization": f"Bearer {self._api_token}",
+                        "Content-Type": "application/json",
+                    },
+                ) as resp:
+                    response = await resp.json()
+                    embeddings.extend(response["embeddings"])
+        return embeddings
+
+    def _get_query_embedding(self, query: str) -> List[float]:
+        """
+        Get query embedding.
+        """
+        return self._post(self._add_query_prefix([query]))[0]
+
+    async def _aget_query_embedding(self, query: str) -> List[float]:
+        """
+        Async get query embedding.
+        """
+        response = await self._apost(self._add_query_prefix([query]))
+        return response[0]
+
+    def _get_query_embeddings(self, queries: List[str]) -> List[List[float]]:
+        """
+        Get query embeddings.
+        """
+        return self._post(self._add_query_prefix(queries))
+
+    async def _aget_query_embeddings(self, queries: List[str]) -> List[List[float]]:
+        """
+        Async get query embeddings.
+        """
+        return await self._apost(self._add_query_prefix(queries))
+
+    def _get_text_embedding(self, text: str) -> List[float]:
+        """
+        Get text embedding.
+        """
+        return self._post(self._add_text_prefix([text]))[0]
+
+    async def _aget_text_embedding(self, text: str) -> List[float]:
+        """
+        Async get text embedding.
+        """
+        response = await self._apost(self._add_text_prefix([text]))
+        return response[0]
+
+    def _get_text_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """
+        Get text embedding.
+        """
+        return self._post(self._add_text_prefix(texts))
+
+    async def _aget_text_embeddings(self, texts: List[str]) -> List[List[float]]:
+        """
+        Async get text embeddings.
+        """
+        return await self._apost(self._add_text_prefix(texts))
+
+    def _add_query_prefix(self, queries: List[str]) -> List[str]:
+        """
+        Add query prefix to queries.
+        """
+        return [self._query_prefix + query for query in queries]
```

Comment:
You can do `[self._query_prefix + query for query in queries] if self._query_prefix else queries` so it doesn't slow things down without query prefix